chainio/txmgr/simple.go

```diff
@@ -91,6 +93,36 @@ func (m *SimpleTxManager) Send(
 	return receipt, nil
 }
 
+// SendWithRetry is used to send a transaction to the Ethereum node, same as Send but adding retry logic.
+// If the transaction fails, it will retry sending the transaction until it gets a receipt, using
+// **exponential backoff** with factor `multiplier`, starting with `initialInterval`.
+func (m *SimpleTxManager) SendWithRetry(
+	ctx context.Context,
+	tx *types.Transaction,
+	initialInterval time.Duration,
+	maxElapsedTime time.Duration,
+	multiplier float64,
+) (*types.Receipt, error) {
+	backoffConfig := backoff.NewExponentialBackOff(
+		backoff.WithInitialInterval(initialInterval),
+		backoff.WithMultiplier(multiplier),
+		backoff.WithMaxElapsedTime(maxElapsedTime),
+	)
+
+	sendAndWait := func() (*types.Receipt, error) {
+		r, err := m.send(ctx, tx)
+		if err != nil {
+			return nil, err
+		}
+		m.logger.Error("failed to send transaction", err)
```

Comment:
think this might be better at a warn level, and might be good to add context about the retry count as well

Reply:
Done